src/main/kotlin/app/revanced/patches/youtube/misc/video/speed/remember/patch/RememberPlaybackSpeedPatch.kt

```diff
@@ -18,120 +18,123 @@ import app.revanced.patches.shared.settings.preference.impl.StringResource
+@RememberPlaybackSpeedCompatibility
 @Version("0.0.1")
-class RememberPlaybackRatePatch : BytecodePatch(
-    listOf(ChangePlaybackRateFragmentStateFingerprint)
+class RememberPlaybackSpeedPatch : BytecodePatch(
+    listOf(ChangePlaybackSpeedFragmentStateFingerprint)
 ) {
-    private companion object {
-        const val INTEGRATIONS_CLASS_DESCRIPTOR =
-            "Lapp/revanced/integrations/patches/playback/speed/RememberPlaybackRatePatch;"
-
-        fun MethodFingerprint.getReference(offsetFromPatternScanResultStartIndex: Int = 0) = this.result!!.let {
-            val referenceInstruction = it.mutableMethod
-                .instruction(it.scanResult.patternScanResult!!.startIndex + offsetFromPatternScanResultStartIndex) as ReferenceInstruction
-            referenceInstruction.reference.toString()
-        }
-
-        fun BytecodeContext.resolveFingerprints() {
-            ChangePlaybackRateFragmentStateFingerprint.result?.also {
-                fun MethodFingerprint.resolve() = resolve(this@resolveFingerprints, it.classDef)
-
-                OnPlaybackRateItemClickFingerprint.resolve()
-                InitializePlaybackRateValuesFingerprint.resolve()
-
-            } ?: throw ChangePlaybackRateFragmentStateFingerprint.toErrorResult()
-        }
-    }
-
     override fun execute(context: BytecodeContext): PatchResult {
         SettingsPatch.PreferenceScreen.MISC.addPreferences(
             SwitchPreference(
-                "revanced_remember_playback_rate_last_selected",
-                StringResource("revanced_remember_playback_rate_last_selected_title", "Remember playback rate changes"),
+                "revanced_remember_playback_speed_last_selected",
+                StringResource(
+                    "revanced_remember_playback_speed_last_selected_title",
+                    "Remember playback speed changes"
+                ),
                 true,
                 StringResource(
-                    "revanced_remember_playback_rate_last_selected_summary_on",
-                    "Playback rate changes apply to all videos"
+                    "revanced_remember_playback_speed_last_selected_summary_on",
+                    "Playback speed changes apply to all videos"
                 ),
                 StringResource(
-                    "revanced_remember_playback_rate_last_selected_summary_off",
-                    "Playback rate changes only apply to the current video"
+                    "revanced_remember_playback_speed_last_selected_summary_off",
+                    "Playback speed changes only apply to the current video"
                 )
             )
         )
 
         context.resolveFingerprints()
 
-        // Set the remembered playback rate.
-        InitializePlaybackRateValuesFingerprint.result!!.apply {
-            // Infer everything necessary for setPlaybackRate()
+        VideoIdPatch.injectCall("${INTEGRATIONS_CLASS_DESCRIPTOR}->newVideoLoaded(Ljava/lang/String;)V")
```

Comment:
Why is this required? The integrations just load the saved playback speed when loading a new video and then the speed is returned at the injection point. Can't the setting simply be returned?

Reply:
> then it needs to know when a new video is loaded so it can reset the video speed back to the default.

There is another injection point which is "getCurrentPlaybackSpeed". Why can't you just return the remembered speed there?